### PR TITLE
Update govuk.buildProject to tag docker similar to github tags

### PR DESCRIPTION
**Pull request by kevindew**
This stops docker tags being created for deployed-to-* branches (as we intend to create those automatically at deploy time) and adds a tag for release which is consistent with our github tags.

This is to enable the tags to be consistent with those which are input into deploy jobs as the deploy jobs will copy the tag inputted to them to create deployed-to-* branches.

### CODE REVIEW

File: modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy [hunk 1]
```diff
@@ -347,6 +346,20 @@ def isCurrentCommitOnMaster() {
   ) == 0
 }
 
+/**
+ * Check whether there is a git branch named release
+ * This test is useful for determining whether we should update this branch or
+ * not
+ */
+def releaseBranchExists() {
+  sshagent(["govuk-ci-ssh-key"]) {
```
Comment (suzannehamilton) [inline modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy]: Does this `sshagent` wrapper return the boolean from lines 356-359? Or does it return some value of its own? I can't find it in the docs.

Comment (kevindew) [reply modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy]: That's a good point. I wasn't too sure and in my testing it worked successfully but I may have only testing the truthy route. I'll give it a spin on something where it is falsy to check.

Comment (kevindew) [reply modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy]: @suzannehamilton Yup I've been able to confirm it returns a boolean: https://ci.integration.publishing.service.gov.uk/job/govuk_app_config/job/master/26/console

**Comment by kevindew** [comment]
Thanks @suzannehamilton 👍

**Pull request merged**